azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -121,41 +125,51 @@ public ExecutorManager(Props props, ExecutorLoader loader,
     executingManager.start();
 
     if(isMultiExecutorMode()) {
-      queueProcessor =
-        new QueueProcessorThread(azkProps.getBoolean(
-          AZKABAN_QUEUEPROCESSING_ENABLED, true), azkProps.getLong(
-          AZKABAN_ACTIVE_EXECUTOR_REFRESHINTERVAL_IN_MS, 1000),
-          azkProps
-            .getInt(AZKABAN_ACTIVE_EXECUTOR_REFRESHINTERVAL_IN_FLOW, 1000));
-      queueProcessor.start();
+      setupMultiExecutorMode();
     }
 
     long executionLogsRetentionMs =
       props.getLong("execution.logs.retention.ms",
         DEFAULT_EXECUTION_LOGS_RETENTION_MS);
 
-    queuedFlows =
-      new QueuedExecutions(props.getLong(AZKABAN_WEBSERVER_QUEUE_SIZE, 100000));
-
     cleanerThread = new CleanerThread(executionLogsRetentionMs);
     cleanerThread.start();
 
   }
 
+  private void setupMultiExecutorMode() {
+    // initliatize hard filters for executor selector from azkaban.properties
+    String filters = azkProps.getString(AZKABAN_EXECUTOR_SELECTOR_FILTERS, "");
+    if (filters != null) {
+      filterList = Arrays.asList(StringUtils.split(filters, ","));
+    }
+
+    // initliatize comparator feature weights for executor selector from
+    // azkaban.properties
+    Map<String, String> compListStrings =
+      azkProps.getMapByPrefix(AZKABAN_EXECUTOR_SELECTOR_COMPARATOR_PREFIX);
+    if (compListStrings != null) {
+      comparatorWeightsMap = new TreeMap<String, Integer>();
+      for (Map.Entry<String, String> entry : compListStrings.entrySet()) {
+        comparatorWeightsMap.put(entry.getKey(), Integer.valueOf(entry.getValue()));
+      }
+    }
+
+    // configure queue processor
+    queueProcessor =
+      new QueueProcessorThread(azkProps.getBoolean(
+        AZKABAN_QUEUEPROCESSING_ENABLED, true), azkProps.getLong(
+        AZKABAN_ACTIVE_EXECUTOR_REFRESHINTERVAL_IN_MS, 1000), azkProps.getInt(
+        AZKABAN_ACTIVE_EXECUTOR_REFRESHINTERVAL_IN_FLOW, 1000));
```

Comment:
should we have MS in the key name as well so it is consistent with AZKABAN_ACTIVE_EXECUTOR_REFRESHINTERVAL_IN_MS?